### PR TITLE
UCP/RMA: refactoring of ucp_rkey_pack function

### PR DESCRIPTION
**Pull request by hoopoepg**
- ucp_rkey_pack function is split to 2 calls to allow use
  serialization of rkeys from another calls

### CODE REVIEW

File: src/ucp/core/ucp_rkey.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+#include <ucp/api/ucp_def.h>
+#include <ucp/core/ucp_ep.h>
+#include <uct/api/uct.h>
+#include <ucs/arch/bitops.h>
+#include <ucs/debug/log.h>
+
+#include <inttypes.h>
+
+ucs_status_t ucp_rkey_write(ucp_context_h context,
```
Comment (yosefe) [inline src/ucp/core/ucp_rkey.h]: this could go to ucp_mm.h

File: src/ucp/core/ucp_rkey.c [hunk 2]
```diff
@@ -18,14 +19,58 @@ static ucp_rkey_t ucp_mem_dummy_rkey = {
 
 static ucp_md_map_t ucp_mem_dummy_buffer = 0;
 
+
+ucs_status_t ucp_rkey_write(ucp_context_h context,
```
Comment (yosefe) [inline src/ucp/core/ucp_rkey.c]: maybe call it ucp_rkey_pack_uct() or something similar?

File: src/ucp/core/ucp_rkey.c [hunk 3]
```diff
@@ -18,14 +19,58 @@ static ucp_rkey_t ucp_mem_dummy_rkey = {
...
+                            ucp_md_map_t md_map, uct_mem_h *memh,
```
Comment (yosefe) [inline src/ucp/core/ucp_rkey.c]: could use `const uct_mem_h *memh`

File: src/ucp/core/ucp_rkey.c (hunk 2 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_rkey.c]: renamed

File: src/ucp/core/ucp_rkey.c (hunk 3 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_rkey.c]: added const modifier

File: src/ucp/core/ucp_rkey.h (hunk 1 above)
Comment (hoopoepg) [reply src/ucp/core/ucp_rkey.h]: done, unused header is removed

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5394/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5395/ for details (Mellanox internal link).

**Pull request merged**